app/init/draft_upload_manager.ts

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {ClientResponse, ClientResponseError} from '@mattermost/react-native-network-client';
+import {AppState, AppStateStatus, DeviceEventEmitter} from 'react-native';
+
+import {updateDraftFile} from '@actions/local/draft';
+import {uploadFile} from '@actions/remote/file';
+import {Events} from '@constants';
+import {PROGRESS_TIME_TO_STORE} from '@constants/files';
+
+type FileHandler = {
+    [clientId: string]: {
+        cancel?: () => void;
+        fileInfo: FileInfo;
+        serverUrl: string;
+        channelId: string;
+        rootId: string;
+        lastTimeStored: number;
+    };
+}
+
+class DraftUploadManager {
+    private handlers: FileHandler = {};
+    private previousAppState: AppStateStatus;
+
+    constructor() {
+        this.previousAppState = AppState.currentState;
+        AppState.addEventListener('change', this.onAppStateChange);
```

Comment:
The listener should be added in the componentDidMount lifecycle method.

Reply:
DraftUploadManager is a simple class, not a react component, so no `componentDidMount` lifecycle.

Reply:
You're right.  I got my thoughts mixed up.